### ISSUE TITLE
Issue with `npm run build`

### ISSUE THREAD

**Issue by imralav** [NevinAF/bga-ts-template#1]
I am having issues integrating `bga-ts-template` into my project. The part of `npm run build` (either invoked directly or at the end of `npm run init` command) always fails for me with the following set of errors:
```
/somepath/BGAStudio/imralavreversi/node_modules/.bin/bga-build: 1: //#region: not found
/somepath/BGAStudio/imralavreversi/node_modules/.bin/bga-build: 2: use strict: not found
: not foundsomepath/BGAStudio/imralavreversi/node_modules/.bin/bga-build: 2:
: not foundomepath/BGAStudio/imralavreversi/node_modules/.bin/bga-build: 3:
/somepath/BGAStudio/imralavreversi/node_modules/.bin/bga-build: 4: Syntax error: "(" unexpected
```

It's as if NPX doesn't recognize the `build.js` from `bga-ts-template` as runnable script.

Have you encountered this before? If so, how to address this?

Best regards
Tomasz

**Comment by imralav** [comment]
I figured that it might be related to my setup, as I am running windows with Ubuntu on WSL, but on windows folder mounted inside WSL. The root problem was that `build.js` had CRLF for new line instead of just LF.

This issue does not exist for other dependencies also downloaded during `bga-ts-template` setup.

The other problem is missing `shebang` of `#!/usr/bin/env node` at the top.

 It started working after did manual changes to the `build.js` file in `node_modules/`

**Comment by NevinAF** [comment]
Good catch and sorry for the trouble!

All of the issues you pointed out are correct. Usually, npm will fix all line endings automatically. Still, it failed to do so because the js files were being called directly AND the file was being automatically generated after the install. In addition, I didn't add the `shebang` to the auto-generated build file which does not help the issue.

Let me know if you have any issues

---

## Fix Version 1.1.7

Commit: 29ce80cbdc51b17d177c73810465815762da383c

### Tested on:
- WSL 2, Ubuntu 22.04, node v20.12.2, npm v10.5.2
- Windows 10, node v18.13.0, npm v8.19.3

### Notes

The build and init commands were added as a runnable module file to fix this. Now, the build and init command files are never touched but instead, the included files are modified. This is how many packages like typescript (tsc) solve this issue.  The package.json file was updated to match these new build scripts.

Snippets from commit:

`package.json`
``` js
{
    "bin": {
        "bga-init": "./bin/bga-init.js",
        "bga-build": "./bin/bga-build.js"
    },
}
```

`bin/bga-build.js`
```js
#!/usr/bin/env node
require('../build.js')
```

`bin/bga-init.js`
```js
#!/usr/bin/env node
require('../initialize.js')
```

**Comment by imralav** [comment]
No need to apologize, this project is amazing, papa bless, it just needs some fine-tuning. I confirm that it for my dev setup.
